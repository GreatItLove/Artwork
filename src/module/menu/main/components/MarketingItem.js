import React from 'react';
import { ListItem } from 'material-ui/List';
import { NavLink } from 'react-router-dom';
import PropTypes from 'prop-types';

import { MARKETING_PATH } from '../../../../config/routesPatch';

const propTypes = {
  style: PropTypes.object.isRequired,
  styleChild: PropTypes.object.isRequired
};

/**
 * @function MarketingItem
 * @param {Object} style
 * @param {Object} styleChild
 * @returns {XML}
 */
function MarketingItem({ style, styleChild }) {
  return (
    <ListItem
      key={MARKETING_PATH}
      primaryText="Marketing"
      innerDivStyle={style}
      leftIcon={<i className="material-icons">public</i>}
      primaryTogglesNestedList={true}
      nestedItems={[
        <ListItem
          key="BulkEmail"
          innerDivStyle={styleChild}
          primaryText="Bulk Email"
          containerElement={<NavLink to={MARKETING_PATH} exact />}
        />,
        <ListItem key="SearchEngineOptimization" innerDivStyle={styleChild} primaryText="Search Engine Optimization" />,
        <ListItem key="WebsiteVisitors" innerDivStyle={styleChild} primaryText="Website Visitors" />
      ]}
    />
  );
}

MarketingItem.propTypes = propTypes;

export default MarketingItem;
