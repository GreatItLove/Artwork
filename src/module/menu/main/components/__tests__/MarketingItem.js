import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';

import MarketingItem from '../MarketingItem';

describe('MarketingItem', () => {
  it('Snapshot of MarketingItem', () => {
    const wrapper = shallow(<MarketingItem style={{}} styleChild={{}} />);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });
});
