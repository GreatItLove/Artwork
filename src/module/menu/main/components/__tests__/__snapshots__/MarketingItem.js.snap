// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MarketingItem Snapshot of MarketingItem 1`] = `
<ListItem
  autoGenerateNestedIndicator={true}
  containerElement="span"
  disableKeyboardFocus={false}
  disabled={false}
  initiallyOpen={false}
  innerDivStyle={Object {}}
  insetChildren={false}
  leftIcon={
    <i
      className="material-icons"
    >
      public
    </i>
  }
  nestedItems={
    Array [
      <ListItem
        autoGenerateNestedIndicator={true}
        containerElement={
            <NavLink
                activeClassName="active"
                ariaCurrent="true"
                exact={true}
                to="/marketing"
            />
        }
        disableKeyboardFocus={false}
        disabled={false}
        initiallyOpen={false}
        innerDivStyle={Object {}}
        insetChildren={false}
        nestedItems={Array []}
        nestedLevel={0}
        onKeyboardFocus={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        onNestedListToggle={[Function]}
        onTouchEnd={[Function]}
        onTouchStart={[Function]}
        open={null}
        primaryText="Bulk Email"
        primaryTogglesNestedList={false}
        secondaryTextLines={1}
    />,
      <ListItem
        autoGenerateNestedIndicator={true}
        containerElement="span"
        disableKeyboardFocus={false}
        disabled={false}
        initiallyOpen={false}
        innerDivStyle={Object {}}
        insetChildren={false}
        nestedItems={Array []}
        nestedLevel={0}
        onKeyboardFocus={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        onNestedListToggle={[Function]}
        onTouchEnd={[Function]}
        onTouchStart={[Function]}
        open={null}
        primaryText="Search Engine Optimization"
        primaryTogglesNestedList={false}
        secondaryTextLines={1}
    />,
      <ListItem
        autoGenerateNestedIndicator={true}
        containerElement="span"
        disableKeyboardFocus={false}
        disabled={false}
        initiallyOpen={false}
        innerDivStyle={Object {}}
        insetChildren={false}
        nestedItems={Array []}
        nestedLevel={0}
        onKeyboardFocus={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        onNestedListToggle={[Function]}
        onTouchEnd={[Function]}
        onTouchStart={[Function]}
        open={null}
        primaryText="Website Visitors"
        primaryTogglesNestedList={false}
        secondaryTextLines={1}
    />,
    ]
  }
  nestedLevel={0}
  onKeyboardFocus={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  onNestedListToggle={[Function]}
  onTouchEnd={[Function]}
  onTouchStart={[Function]}
  open={null}
  primaryText="Marketing"
  primaryTogglesNestedList={true}
  secondaryTextLines={1}
/>
`;
